### ISSUE TITLE
Sprint 1.36.0 2/2 - May 16 to May 27

### ISSUE THREAD

**Issue by marcushyett-ph** [PostHog/posthog#9711]
# Global Sprint Planning

Notes for this coming sprint:
* Changes to company priorities (Quality, ~Extensibility~ Self-serve and deployments)
* We need to minimize work in progress we had too many diverse goals in the previous sprint

## Retro: Status of Outcomes from Previous Sprint

1. Have a team on Cloud successfully running the full persons/groups on events setup (@yakkomajuri) (**status**: Team 7964's new events all contain persons and groups data. We will run a backfill before Friday.)
    * Why? To speed up querying for our biggest customers
1. Have great demo data live on demo.posthog.com (@Twixes) (**status**: almost there, should be ready at the end of the week [famous last words])
      - why: to increase sign ups
1.  Complete nail insights RFC implementations https://github.com/PostHog/product-internal/pull/290 (@mariusandra  @pauldambra ) (**status**: solid progress, most things done, few remaining bugs and testing)
      - why: m2m dashboard: customer requests
1. Frontend "Apps" Support Marketing launch(including notification banner app) (@mariusandra) (**status**: changed scope to github star sync plugin stats app, on track with that)
     - why: Extensibility
1. Cohort filtering implementation (@EDsCODE)  (**status**: Deploying behind feature flag)
    - why: lifecycle capabilities for cohorts are requested often as features
1. Get statsig increase in conversion from signup to first event ingested on onboarding experiment (@liyiy) (**status**: experiment launched on 5/10, will wait about a week for results. Onboarding 1.5 to begin next sprint)
     - why: so we can have more customers be successful with PostHog
1. Get end-to-end persons on events querying working for a single team for Trends (@EDsCODE  TBC) (**status**: Progress made on implementing trends querying)
    - Why? To speed up querying for our biggest customers
1. Complete the work to build an environment completely from scratch and automated (using code, no clicks on a UI). Target is to end up with Heroku (PostgreSQL + Kafka) and AWS resources (EKS with Helm Chart for PostHog) for everything else.  @guidoiaquinti @hazzadous 
    - Why? It's a big step forward/milestone for our 6 months team goals.
1. Team Platform (@fuziontech ): in-place upgrade to get ClickHouse 21.11.1 in PROD (finally, hopefully!).  (**status**: ⚠️ - on track but code: yellow. Sync with @macobo and @rcmarron to get all blockers resolved today )
    - Why? We want to get CH feature/bugfix

## Plan: Proposed Goals for Next Sprint 

Each goal should have a single owner. Owner can only be an engineer.

1. Have the plugin server split between ingestion and runner servers live on Cloud @macobo
Why: To make the plugin server more reliable and easier to manage.
2. Eliminate all legacy ingestion code from the plugin server @tiina303
Why: To reduce tech debt and make the plugin server easier to maintain.
3. Having a dev / staging / prod environments setup and running in new AWS account using EKS in parallel with existing ECS cluster. All stateful resources will be shared between the two environments (Kafka, PG, CH, Redis) but traffic will be able to be consumed from new environment @guidoiaquinti & @hazzadous
why: to move towards our 6 month goals of having infrastructure as code
4. Generate a PNG/SVG/PDF export for any chart @benjackwhite 
why: So we can have better sharing links + subscribe to dashboards
5. Start design/product work on dashboard/insight subscriptions @benjackwhite 
why: so we can build subscribing to dashboards once technical hurdles have been cleared
6. Onboarding 1.5, increase the % of people that ingest events @liyiy 
why: so more people can be successful with PostHog
7. Persons on events querying for trends, but goal is all insights @neilkakkar 
why: To improve performance for biggest customers
9. Async querying @rcmarron 
why: 4% of queries time out, this will allow those queries to complete

# Team sprint planning

For your team sprint planning copy this template into a comment below for each team.

```
Team ___

## Retro

<!-- Talk about what went well, what didn't go well and any actions to improve next time -->

- 

## Hang over items from previous sprint

<!-- For each item, decide to re-prioritise (and add below) or deprioritise -->

- Item 1. prioritised/deprioritise

## Planning

<!-- Each item should have a single owner. Owner can only be an engineer. -->

### High priority

-

### Bugs

-

### Low priority / side quests

-

```

**Comment by mariusandra** [comment]
Team App East

## Retro

- @benjackwhite First sprint. Quite chaotic, bit disruptive. Personally good.
- @Twixes Pretty happy shipping things. Funnels, color scheme, Lemon UI. Demo going a bit slower, very close to useful. Will not have session recordings and other advanced things. Shippable by end of week.
- @pauldambra Happy with progress with dashboards/insights. Met at the hacker house, but didn't make much progress. Support week has been fun (Permanent Wednesday). Found things better tests could have fixed, shame no tests.
- @mariusandra Got covid, out for a few days. Solid progress on frontend apps, hope to have another reviewable PR by the end of the day.
- @lottiecoxon Not much work for this team. Holiday and sick. All over the place. The process for product design is not that clear, will work together close with Chris to get the process nailed. 
- @marcushyett-ph Was great to meet in the Hacker House. Happy to have met Ben.
- @timgl 🦀 

## Hang over items from previous sprint

We will spend a bit of the first week finishing any open issues.

- Frontend Apps @mariusandra - probably a bit of testing still next week
- Insight editor panels @benjackwhite - we will see how far we get
- Insight dashboard many-to-many @pauldambra - still open bugs and testing if anything
- Recordings to S3 @pauldambra - priority this print, as potential data store for images

## Planning

RFC: [Sharing, Exporting, Subscribing Alerting Product Context](https://github.com/PostHog/product-internal/pull/308)

Team goal: 
- Generate a PNG/SVG/PDF export for any chart
- Start design/product work on dashboard/insight subscriptions

### High priority

- Insight exports

### Bugs

- https://github.com/PostHog/posthog/issues/9668
- https://github.com/PostHog/posthog/issues/9666
- https://github.com/PostHog/posthog/issues/9663
- https://github.com/PostHog/posthog/issues/9570
- https://github.com/PostHog/posthog/issues/9568
- https://github.com/PostHog/posthog/issues/9565
- https://github.com/PostHog/posthog/issues/9526
- https://github.com/PostHog/posthog/issues/9508
- https://github.com/PostHog/posthog/issues/9494
- https://github.com/PostHog/posthog/issues/9415

### Low priority / side quests

- [Picking dates](https://github.com/PostHog/product-internal/pull/305)

**Comment by fuziontech** [comment]
Team Platform **Infra**

## Retro

- @guidoiaquinti Not super well TBH. A lot of interruptions and reactive tasks (support, bugfixes). A lot of progress in the grand scheme of things. We are under provisioned and need help.
- @hazzadous Not so much support. Been alright overall. No complaints. Not as much progress as we'd like but good overall.
- @fuziontech - More organizational fires than anticipated. I feel a lot better about the ingestion scheme now that we have Michael and Karl helping out. Escalated the priority for hiring on that team. Clickhouse cluster is still on track but PR around materialized columns was more involved than anticipated. Root cause there was no one person had all of the context on the current state of the world across cloud and on prem for MatCols.

## Hang over items from previous sprint

- Nothing 🥳 

## Planning

### High priority

- Having a dev / staging / prod environments setup and running in new AWS account using EKS in parallel with existing ECS cluster. All stateful resources will be shared between the two environments (Kafka, PG, CH, Redis) but traffic will be able to be consumed from new environment @guidoiaquinti & @hazzadous 
- Stretch goal - 1% of traffic going to new environment @guidoiaquinti & @hazzadous 

### Low priority / side quests

- Monitoring improvements with new environment (based on 1% traffic)
- Brown bag session on new infrastructure 🥪 
- Clean up AWS resources and audit spend @fuziontech

**Comment by yakkomajuri** [comment]
# Team Ingestion

## Retro

- @tiina303:
    - Good sprint
    - Happy to get persons/groups on new events done
    - Unforeseen work related to Redis config change & usage across the stack
    - Glad to be able to drive cluster ops session (good pairing)

- @yakkomajuri: 
    - Good sprint
    - Tasks split worked well
    - Collaboration was good when syncs were needed
    - Fewer outages 🎉 
    - Feel more motivated now that we're getting some help
    - Still spent a considerable amount of time on "reactive" tasks (Kafka healthcheck, plugins, instance support - cloud & self-hosted)

## Hang over items from previous sprint

- Buffer (blocked by performance issues)

## Planning

1. Have the plugin server split between ingestion and runner servers live on Cloud @macobo
    - Why: To make the plugin server more reliable and easier to manage.
2. Eliminate all legacy ingestion code from the plugin server @tiina303
    - Why: To reduce tech debt and make the plugin server easier to maintain.

Extras:
#9652

**Comment by marcushyett-ph** [comment]
@benjackwhite @lottiecoxon would love to kick-off the next sprint with a sync pairing session to work out and break down approach around exporting and subscriptions, if that works for you?

**Comment by benjackwhite** [comment]
@marcushyett-ph sounds good. I created an issue (linked above) which covers my initial thoughts and open questions from a design and tech for this scoped part.

We could use team-east standup (and after) on Monday to discuss further

**Comment by EDsCODE** [comment]
## Team West

### Retro
- Onboarding 1.0 complete — [experiment](https://app.posthog.com/experiments/221) live
- Cohort filtering — released behind feature flag
     - cleanup and instrumentation needs to be done

### Planning
- Onboarding 1.5, increase the % of people that ingest events @liyiy 
- Persons on events querying for trends @neilkakkar 
- Async querying @rcmarron

**Comment by marcushyett-ph** [comment]
@EDsCODE as well as the status in the retro, it'd be great to get an understanding how people's sprints went and how they felt about it.

**Comment by EDsCODE** [comment]
@marcushyett-ph yes! We actually are planning to do a retro of sprints separately from standups to evaluate "how things went" rather than "what was done" which is what we primarily focused on during these planning standups
